src/tasks/MonoTargetsTasks/ILStrip/ILStrip.cs

```diff
@@ -81,4 +122,124 @@ private bool StripAssembly(ITaskItem assemblyItem)
+                    Dictionary<int, int> method_body_uses = new Dictionary<int, int>();
+
+                    foreach (MethodDefinitionHandle mdefh in mr.MethodDefinitions)
+                    {
+                        int methodToken = MetadataTokens.GetToken(mr, mdefh);
+                        MethodDefinition mdef = mr.GetMethodDefinition(mdefh);
+                        int rva = mdef.RelativeVirtualAddress;
+
+                        token_to_rva.Add(methodToken, rva);
+
+                        if (method_body_uses.TryGetValue(rva, out var count))
+                        {
+                            method_body_uses[rva]++;
+                        }
+                        else
+                        {
+                            method_body_uses.Add(rva, 1);
+                        }
+                    }
+
+                    do
+                    {
+                        int methodToken2Trim = Convert.ToInt32(line);
+                        int rva2Trim = token_to_rva[methodToken2Trim];
+                        method_body_uses[rva2Trim]--;
+                    } while ((line = sr.ReadLine()) != null);
+
+                    foreach (var kvp in method_body_uses)
+                    {
+                        int rva = kvp.Key;
+                        int count = kvp.Value;
+                        if (count == 0)
+                        {
+                            MethodBodyBlock mb = peReader.GetMethodBody(rva);
+                            int methodSize = mb.Size;
+                            int sectionIndex = peReader.PEHeaders.GetContainingSectionIndex(rva);
+                            int relativeOffset = rva - peReader.PEHeaders.SectionHeaders[sectionIndex].VirtualAddress;
+                            int actualLoc = peReader.PEHeaders.SectionHeaders[sectionIndex].PointerToRawData + relativeOffset;
+
+                            byte[] zeroBuffer;
+                            zeroBuffer = ArrayPool<byte>.Shared.Rent(methodSize);
+                            for (int i = 0; i < zeroBuffer.Length; i++)
+                            {
+                                zeroBuffer[i] = 0x0;
+                            }
+
+                            memStream.Position = actualLoc;
+                            int firstbyte = memStream.ReadByte();
+                            int headerFlag = firstbyte & 0b11;
+                            int headerSize = headerFlag == 2 ? 1 : 4;
+
+                            memStream.Position = actualLoc + headerSize;
+                            memStream.Write(zeroBuffer, 0, methodSize - headerSize);
+
+                            ArrayPool<byte>.Shared.Return(zeroBuffer);
+                        }
+                    }
+                    memStream.Position = 0;
+                    memStream.CopyTo(os);
+                }
+
+                File.Move(assemblyFilePath, newName_assemblyFilePath);
+                File.Move(trimmedAssemblyFilePath, assemblyFilePath);
```

Comment:
@radical How do we feel about tasks that in-place replace their inputs? I seem to recall the XA "XABT" task used to have a lot of parallelism problems when they in-place replaced mscorlib, but I don't remember the details.